apps/vs-code-designer/src/app/utils/delay.ts

```diff
@@ -4,5 +4,9 @@
  *--------------------------------------------------------------------------------------------*/
 
 export async function delay(ms: number): Promise<void> {
-  await new Promise<void>((resolve: () => void): NodeJS.Timer => setTimeout(resolve, ms));
+  return await new Promise<void>((_, reject) => {
+    setTimeout(() => {
+      reject(new Error());
```

Comment:
Is there any particular reason on why we are rejecting instead of resolving after the timeout?

Reply:
yep, the parent function calling this timeout will catch the error.

Reply:
thinking on it...the name "delay" would be confusing.

Reply:
Yep, thats confusing because its throwing an error everytime after the time. Also, there is an already other delay function.
Also I just noticed that you are updating the function, however I couldn't notice where are you using this function in these changes